backend/shared/shared/audit_log/viewsets.py

```diff
@@ -0,0 +1,104 @@
+from copy import copy
+from typing import Optional, Union
+
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import AnonymousUser
+from django.db.models import Model
+from django.db.models.base import ModelBase
+from rest_framework.viewsets import ModelViewSet
+from shared.audit_log import audit_logging
+from shared.audit_log.enums import Operation, Status
+
+User = get_user_model()
+
+
+class AuditLoggingModelViewSet(ModelViewSet):
```

Comment:
Very nice to have this in the shared code :-)
Requirements for audit logging in Helsinki benefit are still unclear but this should be very useful :-)